python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -421,10 +432,13 @@ class NearestNeighbors(Base):
 
         if self._should_downcast:
             warnings.warn("Parameter should_downcast is deprecated, use "
```

Comment:
Won't this be set to true even if the class was initialized with _should_downcast=None? Because then you assign self._should_downcast to true in the initializer.